src-tauri/src/telemetry_manager.rs

```diff
@@ -0,0 +1,321 @@
+            Network::Esmeralda => TelemetryNetwork::Esmeralda,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum TelemetryMiningMode {
+    Eco,
+    Ludicrous,
+}
+
+impl From<MiningMode> for TelemetryMiningMode {
+    fn from(value: MiningMode) -> Self {
+        match value {
+            MiningMode::Eco => TelemetryMiningMode::Eco,
+            MiningMode::Ludicrous => TelemetryMiningMode::Ludicrous,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TelemetryData {
+    pub app_id: String,
+    pub block_height: u64,
+    pub is_mining_active: bool,
+    pub network: Option<TelemetryNetwork>,
+    pub resource_used: TelemetryResource,
+    pub cpu_hash_rate: Option<f64>,
+    pub cpu_utilization: Option<f32>,
+    pub cpu_make: Option<String>,
+    pub gpu_hash_rate: Option<f64>,
+    pub gpu_utilization: Option<f32>,
+    pub gpu_make: Option<String>,
+    pub mode: TelemetryMiningMode,
+}
+
+pub struct TelemetryManager {
+    node_manager: NodeManager,
+    cpu_miner: Arc<RwLock<CpuMiner>>,
+    gpu_miner: Arc<RwLock<GpuMiner>>,
+    config: Arc<RwLock<AppConfig>>,
+    pub cancellation_token: CancellationToken,
+    pub is_collecting_telemetry: bool,
+    node_network: Option<Network>,
+}
+
+impl TelemetryManager {
+    pub fn new(
+        node_manager: NodeManager,
+        cpu_miner: Arc<RwLock<CpuMiner>>,
+        gpu_miner: Arc<RwLock<GpuMiner>>,
+        config: Arc<RwLock<AppConfig>>,
+        network: Option<Network>,
+    ) -> Self {
+        let cancellation_token = CancellationToken::new();
+        Self {
+            node_manager,
+            cpu_miner,
+            gpu_miner,
+            config,
+            cancellation_token,
+            is_collecting_telemetry: false,
+            node_network: network,
+        }
+    }
+
+    pub fn update_network(&mut self, network: Option<Network>) {
+        self.node_network = network;
+    }
+
+    pub async fn initialize(
+        &mut self,
+        airdrop_access_token: Arc<RwLock<Option<String>>>,
+    ) -> Result<()> {
+        info!("Starting telemetry manager");
+        let telemetry_collection_enabled = self.config.read().await.telemetry_collection;
+        self.is_collecting_telemetry = telemetry_collection_enabled;
+        let _ = self
+            .start_telemetry_process(Duration::from_secs(60), airdrop_access_token)
+            .await;
+        Ok(())
+    }
+
+    async fn start_telemetry_process(
+        &mut self,
+        timeout: Duration,
+        airdrop_access_token: Arc<RwLock<Option<String>>>,
+    ) -> Result<(), TelemetryManagerError> {
+        let node_manager = self.node_manager.clone();
+        let cpu_miner = self.cpu_miner.clone();
+        let gpu_miner = self.gpu_miner.clone();
+        let config = self.config.clone();
+        let cancellation_token: CancellationToken = self.cancellation_token.clone();
```

Comment:
I prefer to use Shutdown like the other code. CancellationToken might be a better choice, but I'd prefer it to all be the same tech